lib/pbench/server/api/resources/query_apis/datasets/datasets_contents.py

```diff
@@ -193,8 +196,12 @@ def postprocess(self, es_json: JSON, context: CONTEXT) -> JSON:
                     }
                 ]
             }
-
         """
+        count = int(es_json["hits"]["total"]["value"])
+        if count == 0:
+            self.logger.info("No data returned by Elasticsearch")
+            return ["NOT_FOUND"]
```

Comment:
Sorry; that's not quite what I'd meant. You're returning a JSON payload which is an array with the single string `"NOT_FOUND"` which will be hard to process. We want to return the HTTP status `HTTPStatus.NOT_FOUND`. We usually return a `message` parameter in the error JSON, so something like:

```
    abort(HTTPStatus.NOT_FOUND, message=f"No directory {CONTEXT['parent']} in {CONTEXT['run_id']} contents.")
```